### PR TITLE
fix(curriculum): update front end demo project urls

### LINKED ISSUE #47891: Update front end demo project URL's
Describe the Issue
All the data viz projects were created on our .rocks domain. We just need to replace the codepen URL's with the new addresses in the curriculum files of those five projects.

Here's the URL's:
https://25--5-clock.freecodecamp.rocks
https://drum-machine.freecodecamp.rocks
https://javascript-calculator.freecodecamp.rocks
https://markdown-previewe.freecodecamp.rocks
https://random-quote-machine.freecodecamp.rocks

And the [project files are here](https://github.com/freeCodeCamp/freeCodeCamp/tree/main/curriculum/challenges/english/03-front-end-development-libraries/front-end-development-libraries-projects)

---

This looks like something that can be fixed by "first-time" code contributors to this repository. Here are the files that you should be looking at to work on a fix:

List of files:

https://github.com/freeCodeCamp/freeCodeCamp/tree/main/curriculum/challenges/english/03-front-end-development-libraries/front-end-development-libraries-projects

Please make sure you read our [guidelines for contributing](https://contribute.freecodecamp.org/#/), we prioritize contributors following the instructions in our guides. Join us in our [chat room](https://discord.gg/PRyKn3Vbay) or our [forum](https://forum.freecodecamp.org/c/contributors/3) if you need help contributing; our moderators will guide you through this.

Sometimes we may get more than one pull request. We typically accept the most quality contribution followed by the one that is made first.

Happy contributing.

### PR DESCRIPTION
**Pull request by Anuran12**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #47891

### CODE REVIEW

**Comment by Anuran12** [comment]
Will it count for Hacktoberfest?

**Comment by raisedadead** [comment]
As a new contributor, we encourage you to read our [contributing guidelines](https://contribute.freecodecamp.org).

We expect our contributors to be aware of the process specific to this project. Following the guidelines religiously earns you the respect of fellow maintainers and saves everyone time.

Some examples of this are:

1. Do not edit files directly through GitHub – while you can, it's not a good idea.
2. Make sure you follow the PR checklist and not just tick things off; otherwise, we won't take you seriously.
3. Use the correct way to link issues in the description of the PR by updating the `XXXXXX`. Do not just add issue numbers everywhere and anywhere you feel like.
4. Keep mentions and review requests to a minimum. We understand you are excited about contributing, and our maintainers will get back to you as soon as they get a chance.
5. Do not work directly off your `main` branch - create a new branch for the changes you are working on.

We appreciate you taking the time to help us, and we hope to see more contributions from you.

Happy Contributing.

File: curriculum/challenges/english/03-front-end-development-libraries/front-end-development-libraries-projects/build-a-markdown-previewer.md
Comment (ieahleen) [inline curriculum/challenges/english/03-front-end-development-libraries/front-end-development-libraries-projects/build-a-markdown-previewer.md:11]: this link doesn't work — I mistyped it, but please test what you are doing.

```suggestion
**Objective:** Build an app that is functionally similar to this: <a href="https://markdown-previewer.freecodecamp.rocks/" target="_blank" rel="noopener noreferrer nofollow">https://markdown-previewer.freecodecamp.rocks/</a>.
```

**Review by majestic-owl448 — DISMISSED** [review]

**Comment by Anuran12** [comment]
> As a new contributor, we encourage you to read our [contributing guidelines](https://contribute.freecodecamp.org).
> 
> We expect our contributors to be aware of the process specific to this project. Following the guidelines religiously earns you the respect of fellow maintainers and saves everyone time.
> 
> Some examples of this are:
> 
> 1. Do not edit files directly through GitHub – while you can, it's not a good idea.
> 2. Make sure you follow the PR checklist and not just tick things off; otherwise, we won't take you seriously.
> 3. Use the correct way to link issues in the description of the PR by updating the `XXXXXX`. Do not just add issue numbers everywhere and anywhere you feel like.
> 4. Keep mentions and review requests to a minimum. We understand you are excited about contributing, and our maintainers will get back to you as soon as they get a chance.
> 5. Do not work directly off your `main` branch - create a new branch for the changes you are working on.
> 
> We appreciate you taking the time to help us, and we hope to see more contributions from you.
> 
> Happy Contributing.

Okay. I followed have seen those and followed those. Only i did not have tasted changes locally sorry.

File: curriculum/challenges/english/03-front-end-development-libraries/front-end-development-libraries-projects/build-a-markdown-previewer.md
Comment (Anuran12) [reply curriculum/challenges/english/03-front-end-development-libraries/front-end-development-libraries-projects/build-a-markdown-previewer.md]: > this link doesn't work — I mistyped it, but please test what you are doing.

Sorry... Actually all links were redirecting me then a discord maintainer told me that those links are not deploying yet. That's why... :⁠^⁠)

I changed the link now .. 

Please review it @ieahleen

**Comment by Anuran12** [comment]
@raisedadead @ieahleen  hey.... I tasted that locally and i guess links are okay now...

**Comment by jeremylt** [comment]
> Okay. I followed have seen those and followed those. Only i did not have tasted changes locally sorry.

You missed the PR naming convention part of the guidelines. It should be something like: fix(curriculum): update front end demo project urls

**Comment by moT01** [comment]
The projects are all live so we are good to get this in when it's ready 👍

**Comment by raisedadead** [comment]
> I tested that locally

If you do not mind, Can you share the steps?

**Comment by Anuran12** [comment]
> > I tested that locally
> 
> If you do not mind, Can you share the steps?

Okay tomorrow morning i will create a screen recording and send the link here. It's already late night

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
> Okay tomorrow morning i will create a screen recording and send the link here. It's already late night

That's quite alright. 

Everyone in the open-source world works asynchronously. You don't have to respond to any comment request right away. Learn more: https://www.google.com/search?q=working+asynchronously

I asked for steps, meaning sharing a quick note like "I ran this command and so on.." is usually enough. You don't need to "show-and-tell" when working async :)

I can tell that this PR was not tested locally (by installing freeCodeCamp) because you made every commit from GitHub directly.

My recommendation is to take the contribution guidelines seriously; this helps you earn respect & trust of your fellow maintainers. That's how you one day may become a maintainer for one of the projects out there on GitHub and elsewhere.

Remember, Hacktoberfest is NOT a contest. It's a celebration of open-source, which encourages everyone to learn "collaboration".

You have done an excellent job with the PR :)

Cheers and happy contributing.

**Comment by Anuran12** [comment]
@raisedadead 

Actually i am not home right now so i don't have any access to my computer for the last two days. Tomorrow i will go home. I don't have a laptop but i wanted to contribute consistently, you can see my GitHub profile as i started almost 8 days ago and am trying to contribute everyday. And i am loving to contribute. 

So I have been using my phone for the last two days. And freecodecamp is always a great opportunity for me i also tried before but someone solved that issue before me 🥲. Well, whenever i use my phone i just deploy that on vercel and edit on GitHub. But i was unable to deploy this there.

 I thought tomorrow i will deploy that on localhost on my computer and send you that. 🥲

Sorry i am a newbie so if i break any rule Please forgive me this time. Thanks for your guidance 🙏

**Comment by Anuran12** [comment]
@raisedadead you seems good to me. Can you please guide me more about open source? I have very little knowledge but want to learn lots of things. And next year i want to participate in GSOC. So i need guidance for that 😅

**Comment by raisedadead** [comment]
As I said, do not worry about it. There is no deadline here.

>  have very little knowledge but want to learn lots of things.

Please check out our community forum and chat about these. While some of are unable to help you on a personal level, the members of the community would be more than happy to help you out with questions - coding, career, and anything else.

**Comment by Anuran12** [comment]
@raisedadead  hey will not you merge this ?

**Comment by raisedadead** [comment]
> Keep mentions and review requests to a minimum. We understand you are excited about contributing, and our maintainers will get back to you as soon as they get a chance.

**Comment by Anuran12** [comment]
@ieahleen  can you please review this ?

**Comment by raisedadead** [comment]
@Anuran12, 

We have repeatedly warned you about "@mentioning" and comment on requests. It seems that you are not paying attention. Please remember we have a very strict [code of conduct](https://www.freecodecamp.org/news/code-of-conduct/).

Further violations of this will lead to enforcing it.

**Comment by Anuran12** [comment]
Sorry I misunderstood.... I understood that I have to mention members until it will be merged . but now I understand that I should not mention unnecessarily.

**Review by moT01 — APPROVED** [review]
LGTM @Anuran12 🎉 Congratulations on your first contribution to the freeCodeCamp repo 🎉 We look forward to future contributions

**Pull request merged**
